diff --git a/src/Layouts/ScreencastQueryHandlerType.php b/src/Layouts/ScreencastQueryHandlerType.php
index 783c73b..0f33ace 100644
--- a/src/Layouts/ScreencastQueryHandlerType.php
+++ b/src/Layouts/ScreencastQueryHandlerType.php
@@ -7,7 +7,9 @@ use Netgen\Layouts\API\Values\Collection\Query;
 use Netgen\Layouts\Collection\QueryType\QueryTypeHandlerInterface;
 use Netgen\Layouts\Parameters\ParameterBuilderInterface;
 use Netgen\Layouts\Parameters\ParameterType\TextType;
+use Symfony\Component\DependencyInjection\Attribute\AutoconfigureTag;
 
+#[AutoconfigureTag('netgen_layouts.query_type_handler', ['type' => 'screencast_query'])]
 class ScreencastQueryHandlerType implements QueryTypeHandlerInterface
 {
     public function __construct(
