diff --git a/templates/base.html.twig b/templates/base.html.twig
index 6fc6f97..b00fe50 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -19,6 +19,7 @@
             {% block header %}
                 <nav>
                     <a href="{{ path('app_homepage') }}">Home</a>
+                    <a href="{{ path('app_screencast_index') }}">Screencasts</a>
                     <a href="{{ path('app_contact') }}">Contact Us</a>
                 </nav>
             {% endblock %}
