diff --git a/src/Layouts/ScreencastValueUrlGenerator.php b/src/Layouts/ScreencastValueUrlGenerator.php
index 12d21af..271c86a 100644
--- a/src/Layouts/ScreencastValueUrlGenerator.php
+++ b/src/Layouts/ScreencastValueUrlGenerator.php
@@ -2,12 +2,22 @@
 
 namespace App\Layouts;
 
+use App\Entity\Screencast;
 use Netgen\Layouts\Item\ValueUrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
 
-class ScreencastUrlGenerator implements ValueUrlGeneratorInterface
+class ScreencastValueUrlGenerator implements ValueUrlGeneratorInterface
 {
+    public function __construct(private RouterInterface $router)
+    {
+    }
+
     public function generate(object $object): ?string
     {
-        // TODO: Implement generate() method.
+        assert($object instanceof Screencast);
+
+        return $this->router->generate('app_screencast_show', [
+            'id' => $object->getId()
+        ]);
     }
 }
